### ISSUE TITLE
Improve YAML config validation error messaging

### ISSUE THREAD

**Issue by patduin** [HotelsDotCom/waggle-dance#99]
_As a user of Waggle Dance_
_I'd like to get some meaningful error when the yml config has errors_
_So that I can debug it myself_

The yml config is sort of fiddly at times and when for instance the `federation-server.yml` is not indented correctly it fails to start WD and because logging isn't initialized the user gets no feedback.
Reproduce, try to load this config with (notice the extra spaces on the last two lines):
```primary-meta-store:
  database-prefix: ''
  name: local
  remote-meta-store-uris: thrift://localhost:9083
federated-meta-stores:
- remote-meta-store-uris: thrift://localhost:9083
    database-prefix: remote_
    name: remote
```

Acceptance Criteria:
- A bad yml config file is reject by WD with an error message (the more specific the better but an indication which file is wrong is already much better then current situation)

**Comment by massdosage** [comment]
@patduin Just checking, this has been fixed as a side effect of #105 and can be closed correct?

**Comment by patduin** [comment]
yes
